### PR TITLE
Add strict versioning and site reporting to Java build

### PR DESCRIPTION
**Pull request by karianna**
### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [ ] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Comment by karianna** [comment]
labelling bot could possibly detect Java and do it's thing now.

File: .gitignore [hunk 1]
```diff
@@ -465,5 +465,6 @@ data/qdrant
 data/chatstore*
 
 # Java build
+java/.java-version
 java/**/target
```
Comment (johnoliver) [inline .gitignore:469]: We cant modify files outside of the java directory, if you want changes outside of java, make them on the main branch

File: .vscode/settings.json [hunk 2]
```diff
@@ -74,5 +74,7 @@
   },
```
Comment (johnoliver) [inline .vscode/settings.json:74]: We cant modify files outside of the java directory, if you want changes outside of java, make them on the main branch

**Review by markwallace-microsoft — COMMENTED** [review]
Separate the `settings.json` file into it's own PR.

Otherwise LGTM

**Comment by karianna** [comment]
Well rebase went horribly wrong.

**Pull request closed without merging**